### ISSUE TITLE
Write tag for Darksoft / NeoSD Neogeo games from usb0 doesn't work

### ISSUE THREAD

**Issue by coded-with-claws** [ElRojo/MiSTerRFID#20]
Hi, it's me again! :)

I tried to add Neogeo games since you did the merge request from find-file branch. I hadn't took the time yet to test for Neogeo games. My games are located into `/media/usb0/games/NEOGEO/`.
I encountered two problems:

1. In the mgl file, the path attribute is "None found!". The `/tmp/CURRENTPATH` contains elaborated title names like `Puzzle Bobble (bootleg) (pbbblenb)` or `Windjammers`, hence they can't be found by the search function.
2. If I fix the mgl file but keeping it into `/media/usb0/games/NEOGEO/`, it doesn't work.

Workarounds found:
1. 
I manually edited the mgl file to set the name of the directory containing the game (for example `wjammers`).

The problem is due to the presence of a file `romsets.xml` because I use the Darksoft romset. When I open the `Load ROM set *.*` menu, I have the true titles of the games (`Windjammers`) instead of the directories (`wjammers`). If I remove this `romsets.xml` file, I see the directories, but I can't load anymore the games (the directory appears as being empty).
I think it's likely to be a difficult problem to handle, because the files like `/tmp/CURRENTPATH` don't give us the game's directory name. Maybe a hack into the bash search function...?

I tried to switch to `.neo` files without a `romsets.xml` file, but it still doesn't find the game.
The file I load is `pbobblen.neo` and `/tmp/CURRENTPATH` contains `Puzzle Bobble`, not ending with `.neo` so I guess that's why it doesn't find.

2.
If I move the mgl file to `/media/fat/games/NEOGEO/`, it works, but the path must only contain the game directory's name (for example `wjammers` so that it's agnostic from the drive containing the games:
```
cat /media/fat/games/NEOGEO/wjammers.mgl
<mistergamedescription><rbf>_Console/NeoGeo</rbf><file delay="2" type="f" index="1" path="wjammers"/></mistergamedescription>
```

In conclusion:
1. The search function could be improved to know about the Darksoft romset title names, and also handle the case of absence of `.neo` extension.
2. Seems like we should always write the mgl files to `/media/fat/games/NEOGEO/`.

Maybe you'll encounter these problems when loading from your NAS and have other ideas?

**Comment by coded-with-claws** [comment]
One more last precision for point 2: when I said the mgl file doesn't work from usb0, it loads the NeoGeo core but displays the "cross hatch test", like it doesn't find the game, even though the path into the mgl is correct (path on usb0).

**Comment by ElRojo** [comment]
Thanks for submitting this. As far as I can tell, the mgl files do not work with the Darksoft ROMset: https://misterfpga.org/viewtopic.php?p=60939#p60939

It looks like they are working on a fix for that and I plan to make some updates once that is taken care of. I had to use a different ROMset with `.neo` files for NeoGeo to work.

**Comment by coded-with-claws** [comment]
Thanks for that information, I'll switch my fullset to the .neo one :+1:
